File: Hahn.ApplicationProcess.February2021.Domain/Entities/Asset.cs
using System;

namespace Hahn.ApplicationProcess.February2021.Domain.Entities
{
    public class Asset : BaseEntity
    {

        public string AssetName { get; set; }
        public DepartmentType Department { get; set; }
        public string CountryOfDepartment { get; set; }
        public string EMailAdressOfDepartment { get; set; }
        public DateTimeOffset PurchaseDate { get; set; }
        public bool broken { get; set; }
    }

    public enum DepartmentType
    {
        HQ = 1,
        Store1 = 2,
        Store2 = 3,
        Store3 = 4,
        MaintenanceStation = 5
    }


}

File: hahn-client/src/models/asset.ts
export class Asset
{
    constructor() {
        this.id = 0,
        this.assetName = "",
        this.countryOfDepartment = "",
        this.eMailAdressOfDepartment = "",
        this.department = "",
        this.isBroken = false
    }

    id: number;
    assetName: string;
    department: string;
    countryOfDepartment: string;
    eMailAdressOfDepartment: string;
    purchaseDate: Date;
    isBroken:boolean;

    //gsmith add and test
    public initialAssetLoad()
    {
        let result = this.assetName !== "" &&
        this.eMailAdressOfDepartment !== "" &&
        this.department !== "" &&
        this.countryOfDepartment !== "" &&
        this.purchaseDate !== undefined
        return result;
    }
}